Tests/ComposableArchitectureTests/TestStoreNonExhaustiveTests.swift

```diff
@@ -684,6 +684,71 @@
       await store.receive(/NonExhaustiveReceive.Action.response2)
     }
 
+    func testXCTModifyExhaustive() async {
+      struct State: Equatable {
+        var child: Int? = 0
+        var count = 0
+      }
+      enum Action: Equatable { case tap, response }
+      let store = TestStore(
+        initialState: State(),
+        reducer: Reduce<State, Action> { state, action in
+          switch action {
+          case .tap:
+            state.count += 1
+            return .send(.response)
+          case .response:
+            state.count += 1
+            return .none
+          }
+        }
+      )
+
+      await store.send(.tap) { state in
+        state.count = 1
+        XCTExpectFailure {
+          XCTModify(&state.child, case: /Optional.some) { _ in }
```

Comment:
Minor take-it-or-leave-it, but we have a shorthand syntax for optionals:

```suggestion
          XCTModify(&state.child, case: /.some) { _ in }
```